crates/api/src/site/mod_log.rs

```diff
@@ -98,13 +241,75 @@ impl Perform for GetModlog {
     ) = if data.community_id.is_none() {
       blocking(context.pool(), move |conn| {
         Ok((
-          ModRemoveCommunityView::list(conn, mod_person_id, page, limit)?,
-          ModBanView::list(conn, mod_person_id, page, limit)?,
-          ModAddView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgePersonView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgeCommunityView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgePostView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgeCommentView::list(conn, mod_person_id, page, limit)?,
+          if matches!(
+            type_,
+            ModlogActionType::All | ModlogActionType::ModRemoveCommunity
+          ) && other_person_id.is_none()
+          {
+            ModRemoveCommunityView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+          } else {
+            Vec::<ModRemoveCommunityView>::new()
+          },
+          if matches!(type_, ModlogActionType::All | ModlogActionType::ModBan) {
+            ModBanView::list(
+              conn,
+              mod_person_id,
+              other_person_id,
+              page,
+              limit,
+              hide_modlog_names,
+            )?
+          } else {
+            Vec::<ModBanView>::new()
+          },
+          if matches!(type_, ModlogActionType::All | ModlogActionType::ModAdd) {
+            ModAddView::list(
+              conn,
+              mod_person_id,
+              other_person_id,
+              page,
+              limit,
+              hide_modlog_names,
+            )?
+          } else {
+            Vec::<ModAddView>::new()
+          },
+          if matches!(
+            type_,
+            ModlogActionType::All | ModlogActionType::AdminPurgePerson
+          ) && other_person_id.is_none()
+          {
+            AdminPurgePersonView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+          } else {
+            Vec::<AdminPurgePersonView>::new()
+          },
+          if matches!(
+            type_,
+            ModlogActionType::All | ModlogActionType::AdminPurgeCommunity
+          ) && other_person_id.is_none()
+          {
```

Comment:
This && other_person_id.is_none() is used so much, that it should be another whole if section.

Reply:
Only 4 times in this section, and I can't pull it out due to the way this whole section is written. I can't return 4 lists in a single closure it looks like.

Reply:
Closures can have tuple outputs, you should be able to do:

`(list1, 2 3 4)` and have that be the output of the closure.